app/controllers/katello/api/v2/host_collections_controller.rb

```diff
@@ -138,30 +152,33 @@ def copy
     private
 
     def find_host_collection
-      @organization = @system.organization if @system
       @organization = @activation_key.organization if @activation_key
-      @host_collection = HostCollection.where(:id => params[:id]).first
-      fail HttpErrors::NotFound, _("Couldn't find host collection '%s'") % params[:id] if @host_collection.nil?
+      id = params[:host_collection_id] || params[:id]
+      @host_collection = HostCollection.where(:id => id).first
+      fail HttpErrors::NotFound, _("Couldn't find host collection '%s'") % id if @host_collection.nil?
     end
 
     def host_collection_params
-      attrs = [:name, :description, :max_content_hosts, :unlimited_content_hosts, { :system_ids => [] }]
+      attrs = [:name, :description, :max_hosts, :unlimited_hosts, { :host_ids => [] }]
       params.fetch(:host_collection).permit(*attrs)
     end
 
-    def host_collection_params_with_system_uuids
+    def host_collection_params_with_host_ids
       result = host_collection_params
-      if params['system_uuids']
-        systems_from_uuid = System.uuids_to_ids(params['system_uuids'])
-        result['system_ids'] = result['system_ids'] ?  result['system_ids'] + systems_from_uuid : systems_from_uuid
-      end
-      result[:max_content_hosts] = nil if params[:unlimited_content_hosts]
+      result[:max_hosts] = nil if params[:unlimited_hosts]
       result
     end
 
-    def find_system
-      @system = System.find_by!(:uuid => params[:system_id]) if params[:system_id]
-      @organization = @system.organization if @system
+    def find_host
+      @host = ::Host::Managed.find(params[:host_id]) if params[:host_id]
```

Comment:
This isn't properly checking that the user has edit permissions on the host.  Here's an example where i checked view on hosts:  https://github.com/Katello/katello/pull/5695/files#diff-beaa1cde8d4964406e1f688ae9de5befR71

(and i added that find_host_with_subscriptions method into api_controller, but you don't care about subscriptions, so just check the implementation of that method).